api/app/controllers/review.controller.ts

```diff
@@ -222,3 +226,104 @@ export async function submit(req: Request, res: Response) {
     });
   }
 }
+
+export const getAllReviewsForAdmin = async (req: Request, res: Response) => {
+  const submittedAt = req.query.submittedAt as string;
+  const transcriptId = Number(req.query.transcriptId)
+  const userId = Number(req.query.userId);
+  const mergedAt = req.query.mergedAt as string;
+  const userSearch = req.query.user as string;
+  const page: number = Number(req.query.page) || DB_START_PAGE;
+  const limit: number = Number(req.query.limit) || DB_QUERY_LIMIT;
+
+  const condition: {
+    [key: string | number]: any;
+  } = {};
+
+  const userCondition: {
+    [Op.or]?: {
+      email?: { [Op.iLike]: string };
+      githubUsername?: { [Op.iLike]: string };
+    }[];
+  } = {};
+
+  // Add conditions if query parameters exist
+  if (Boolean(submittedAt)) {
+    condition.submittedAt = {
+      [Op.gte]: new Date(submittedAt as string),
+      [Op.lte]: new Date(submittedAt as string),
+    };
+  }
+  if (Boolean(transcriptId)) {
+    condition.transcriptId = { [Op.eq]: transcriptId };
+  }
+  if (Boolean(userId)) {
+    condition.userId = { [Op.eq]: userId };
+  }
+  if (Boolean(mergedAt)) {
+    condition.mergedAt = {
+      [Op.gte]: new Date(mergedAt as string),
+      [Op.lte]: new Date(mergedAt as string),
+    };
+  }
+
+  if (userSearch) {
+    const searchCondition = { [Op.iLike]: `%${userSearch.toLowerCase()}%` };
+    userCondition[Op.or] = [
+      { email: searchCondition },
+      { githubUsername: searchCondition },
+    ];
+  }
+
+  try {
+    const reviews = await Review.findAll({
+      where: condition,
+      order: [["createdAt", "DESC"]],
+      include: [
+        { model: Transcript, required: true, attributes: { exclude: ["id"] } },
+        {
+          model: User,
+          attributes: { exclude: ["id", "jwt", "albyToken"] },
+          where: userCondition,
+          required: true,
+        },
+      ],
+    });
+
+    const reviewCount = await Review.count({
+      distinct: true,
+      where: condition,
+      include: [
+        { model: Transcript, required: true, attributes: { exclude: ["id"] } },
```

Comment:
we can remove the attributes since its just a count
```suggestion
        { model: Transcript, required: true },
```